test/example_data.dart

```diff
@@ -804,7 +804,7 @@ InitialSnapshot initialSnapshot({
     zulipMergeBase: zulipMergeBase ?? recentZulipVersion,
     alertWords: alertWords ?? ['klaxon'],
     customProfileFields: customProfileFields ?? [],
-    emailAddressVisibility: EmailAddressVisibility.everyone,
+    emailAddressVisibility: emailAddressVisibility ?? EmailAddressVisibility.everyone,
```

Comment:
```
initial_snapshot: Add emailAddressVisibility override in InitialSnapshot method

The InitialSnapshot method does not override the
emailAddressVisibility in the example_data.dart
with the value passed in the function arguments.
This commit fixes the bug
```

Commit message nit: I think the paragraph ("The InitialSnapshot method…") isn't necessary; it just repeats what's already clear in the code change. Instead of that paragraph, how about:

- "It looks like this was left out by accident." (or something like that)
- Say whether or not any callers were passing `emailAddressVisibility`. That helps make clear what changes this is making to our tests' behavior, if any. (From a quick search, I think none of the callers were passing it.)

Reply:
Okay, thanks for the comment, I will update the commit message.

Yes, at the moment no callers were passing `emailAddressVisibility`, but when I was working on the @mention autocomplete items, I was trying to write a test that required passing the `emailAddressVisibility`, that is when I discovered the issue, Even if it is not used now, I think there will be future scenarios where it would be needed.

Reply:
Makes sense. Yes, this is still a good change to make! :) Just that it's helpful to know if it's a change that affects behavior of existing tests. It sounds like it doesn't, and that's helpful to write down in the commit message, for anyone who's reading it who doesn't know that yet.

Reply:
Please post on the PR thread when you've made the change and it's ready for another review.

Reply:
Hi @chrisbobbe 
I have made updates to the commit, please review.